src/test/ui/async-await/async-await-let-else.stderr

```diff
@@ -35,7 +35,8 @@ LL |         bar2(Rc::new(())).await
    |              |
    |              has type `Rc<()>` which is not `Send`
 LL |     };
-   |      - `Rc::new(())` is later dropped here
+LL | }
+   | - `Rc::new(())` is later dropped here
```

Comment:
This seems weird to me. I would expect that `Rc::new(())` would get dropped at the end of the else block, not the end of the function?

Reply:
Sounds right. Indeed this is not expected as `Rc::new(())` should be a temporary value and its lifetime terminates at the end of the `else` block.

`else` block should be marked as enclosed in a terminating scope. This should fix this bug.

Reply:
Okay, now the temporaries are properly contained within the `else` block now.